core/unix/signal_linux_riscv64.c

```diff
@@ -148,10 +149,18 @@ sigill_detected(void *func)
     set_handler_sigact(&act, SIGILL, (handler_t)catch_sigill);
     sigaction_syscall(SIGILL, &act, &old_act);
 
+    /* We use dr_longjmp to exit the SIGILL handle, which skips the signal mask restoring
+     * of OS. Manually save and restore the signal mask here.
+     */
+    kernel_sigset_t oset;
+    dynamorio_syscall(SYS_rt_sigprocmask, 4, SIG_SETMASK, NULL, &oset, sizeof(oset));
```

Comment:
Let's use sigprocmask_syscall to match other signal code: export it from signal.c.  (Let's avoid someone copying this code into cross-UNIX files and having it fail on Mac, e.g..)

Reply:
Currently `sigprocmask_syscall` is declared as `static` in signal.c. To export it, I will remove `static` and redeclare it in signal_private.h.